crates/sui-framework/sources/bcs.move

```diff
@@ -0,0 +1,148 @@
+// Copyright (c) 2022, Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+module sui::bcs {
+    use std::vector as v;
+
+    /// For when bytes length is less than required for deserialization.
+    const EOutOfRange: u64 = 0;
+
+    /// Read address from the bcs-serialized bytes.
+    public fun peel_address(bcs: &mut vector<u8>): address {
+        assert!(v::length(bcs) >= 20, EOutOfRange);
+        let (_value, i) = (0, 0);
+        while (i < 20) {
+            let _ = v::remove(bcs, 0);
+            i = i + 1;
+        };
+        @0x0
+    }
+
+    /// Read a `bool` value from bcs-serialized bytes.
+    public fun peel_bool(bcs: &mut vector<u8>): bool {
```

Comment:
In the original impl, not all `u8` bit patterns are valid `bool`:

https://github.com/diem/bcs/blob/master/src/de.rs#L95

Reply:
I would actually suggest more flexible deserialization (more js-like) so that `bool` can be a check for non-zero value. But for now following the spec.